### PR TITLE
SDK PR for Add enableCrossZoneUpgrade and prioritizeUnhealthyInstances properties to RollingUpgradePolicy for VMSS

### PR DESCRIPTION
**Pull request by kimberlyyipc**
REST API Pull Request: https://github.com/Azure/azure-rest-api-specs/pull/11184

### CODE REVIEW

**Review by markcowl — CHANGES_REQUESTED** [review]
The code changes look OK, but there are some things that are missing:
- Please add in a reference to the corresponding Swagger PR for this change
- running the generation script produces an updated metadata file in the eng directory, please add this in to the PR
- When api versions change, you need to update Az.Sdk.RP.props to change the tag to match the new api-version
- You still need to re-record the tests and add in new tests for any new functionality

Also, can you coordinate with the owner of this PR: https://github.com/Azure/azure-sdk-for-net/pull/16844

**Comment by kimberlyyipc** [comment]
> The code changes look OK, but there are some things that are missing:
> 
> * Please add in a reference to the corresponding Swagger PR for this change
> * running the generation script produces an updated metadata file in the eng directory, please add this in to the PR
> * When api versions change, you need to update Az.Sdk.RP.props to change the tag to match the new api-version
> * You still need to re-record the tests and add in new tests for any new functionality
> 
> Also, can you coordinate with the owner of this PR: #16844

**Comment by kimberlyyipc** [comment]
@markcowl I've 
- added reference to the corresponding Swagger PR for this change 
- added the metadata file in the eng directory
- updated Az.Sdk.RP.props to reflect the new api-version

and cannot record/ work on new tests yet due to ARM manifest rollout

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs [hunk 1]
```diff
@@ -134,7 +138,7 @@ public async Task<AzureOperationResponse<Gallery>> UpdateWithHttpMessagesAsync(s
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, string select = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs:141]: This is a breaking change - we should add an overload in customization that matches the old parameters

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs [hunk 2]
```diff
@@ -150,12 +154,16 @@ public static Gallery Get(this IGalleriesOperations operations, string resourceG
...
             /// </param>
+            /// <param name='select'>
+            /// The select expression to apply on the operation. Possible values include:
+            /// 'Permissions'
+            /// </param>
             /// <param name='cancellationToken'>
             /// The cancellation token.
             /// </param>
-            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, string select = default(string), CancellationToken cancellationToken = default(CancellationToken))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs:164]: same - should add customization to protect from breaking change

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs [hunk 3]
```diff
@@ -48,22 +48,23 @@ public GalleryImageUpdate()
...
+        /// image definition.</param>
+        public GalleryImageUpdate(OperatingSystemTypes osType, OperatingSystemStateTypes osState, GalleryImageIdentifier identifier, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string description = default(string), string eula = default(string), string privacyStatementUri = default(string), string releaseNoteUri = default(string), string hyperVGeneration = default(string), IList<GalleryImageFeature> features = default(IList<GalleryImageFeature>), System.DateTime? endOfLifeDate = default(System.DateTime?), RecommendedMachineConfiguration recommended = default(RecommendedMachineConfiguration), Disallowed disallowed = default(Disallowed), ImagePurchasePlan purchasePlan = default(ImagePurchasePlan), string provisioningState = default(string))
```
Comment (markcowl) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs:67]: This is a breaking change, should add a customization with a constructor overload that matches the old version

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@kimberlyyipc Nted a couple of places where there are breaking changes we should work around.  Otherwise, lgtm

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs (hunk 1 above)
Comment (kimberlyyipc) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs:141]: How is this a breaking change? Isn't this new added parameter optional?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs (hunk 2 above)
Comment (kimberlyyipc) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs:164]: wondering the same thing about this change

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs (hunk 1 above)
Comment (kimberlyyipc) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs:141]: Nvm comment above ^ I can see why it is a breaking change now. I've added some updates for this, please take a look.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs (hunk 2 above)
Comment (kimberlyyipc) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs:164]: resolved

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs (hunk 3 above)
Comment (kimberlyyipc) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs:67]: added

**Comment by kimberlyyipc** [comment]
@markcowl I've added some updates and a new test, please take a look, thank you.

**Review by markcowl — CHANGES_REQUESTED** [review]
You cannot fix breaking changes by editing the generated code.  Any changes you make by hand to generated code will disappear on the next generation.

Instead, Since these are partial classes, you can fix them by adding corresponding partial classes to the 'Customizations' folder, with the necessayr changes.

See this review:  https://github.com/Azure/azure-sdk-for-net/pull/16844

For examples of how to fix breaking changes in method parameters, for example.

**Comment by markcowl** [comment]
@kimberlyyipc Can you please fix the merge conflicts?

**Comment by kimberlyyipc** [comment]
> @kimberlyyipc Can you please fix the merge conflicts?

@markcowl I've resolved the merge conflicts

**Comment by Sandido** [comment]
Closing for now.

**Comment by markcowl** [comment]
@allenjzhang Assigned to you as this is one of the compute changes you are tracking

**Pull request closed without merging**